extensions/data-workspace/src/services/workspaceService.ts

```diff
@@ -229,7 +234,8 @@ export class WorkspaceService implements IWorkspaceService {
 		// so the filter needs to be in the format folder/**/*.sqlproj if there's only one supported projectextension
 		const projFilter = supportedProjectExtensions.length > 1 ? path.posix.join(escapedPath, '**', `*.{${supportedProjectExtensions.toString()}}`) : path.posix.join(escapedPath, '**', `*.${supportedProjectExtensions[0]}`);
 
-		return await glob(projFilter);
+		// glob will return an array of file paths with forward slashes, so they need to be converted back if on windows
+		return (await glob(projFilter)).map(p => folder.fsPath.includes('\\') ? p.split('/').join('\\') : p);
```

Comment:
is this really safe? '\' is a valid character for path on Linux.

Reply:
You should be able to just use path.resolve